eframe/src/web/mod.rs

```diff
@@ -96,13 +98,21 @@ pub fn canvas_size_in_points(canvas_id: &str) -> egui::Vec2 {
 
 pub fn resize_canvas_to_screen_size(canvas_id: &str, max_size_points: egui::Vec2) -> Option<()> {
     let canvas = canvas_element(canvas_id)?;
+    let parent = canvas.parent_element()?;
+
+    let width = parent.scroll_width();
+    let height = parent.scroll_height();
+
+    let canvas_real_size = Vec2 {
+        x: width as f32,
+        y: height as f32,
+    };
```

Comment:
On my Firefox and Chrominum on Mac, this gives a size of `2001 x 0` which makes nothing visible at all.

This should be reverted to the old code (calling `screen_size_in_native_points`), especially as I don't see how this is related to a graceful exit!

Reply:
> On my Firefox and Chrominum on Mac, this gives a size of `2001 x 0` which makes nothing visible at all.

The idea was to control width of canvas by wrapping in by div, and set width of that div manually.



> This should be reverted to the old code (calling `screen_size_in_native_points`), especially as I don't see how this is related to a graceful exit!

Its all about having control of egui from web. To embed egui in div, not full-frame document.

Reply:
That may be, but the new code breaks the the web demo.

Reply:
That' is why i made this https://github.com/emilk/eframe_template/pull/60

Anyway, may be I add fallback to screen_size_in_native_points

Reply:
Oh. Okay. I'll try to change it too

Reply:
It works now ❤️ thanks!